tests/functional_tests/ffront_tests/test_type_deduction.py

```diff
@@ -648,6 +649,33 @@ def tuple_where_mix_dims(
     )
 
 
+def test_astype_dtype():
+    ADim = Dimension("ADim")
+
+    def simple_astype(a: Field[[ADim], float64]):
+        return astype(a, bool)
+
+    parsed = FieldOperatorParser.apply_to_function(simple_astype)
+
+    assert parsed.body[0].value.type == ct.FieldType(
+        dims=[ADim], dtype=ct.ScalarType(kind=ct.ScalarKind.BOOL)
+    )
+
+
+def test_astype_wrong_input():
+    ADim = Dimension("ADim")
+    msg = "Could not deduce type"
+
+    def bad_input_astype(a: Field[[ADim], float64]):
+        return astype(a, "float64")
+
+    with pytest.raises(
+        FieldOperatorTypeDeductionError,
+        match=msg,
```

Comment:
The message should either be more precise or is not needed ("Could not deduce type" is already implied by the type of the error.)